identity/did.go

```diff
@@ -64,6 +150,12 @@ type ServiceDID interface {
 	// ValidateKey checks if a given key is valid for the given centrifugeID.
 	ValidateKey(ctx context.Context, did DID, key []byte, purpose int64) error
 
+	// ValidateSignature checks if signature is valid for given identity
+	ValidateSignature(signature *coredocumentpb.Signature, message []byte) error
```

Comment:
I think this method is not needed. I replaced it with `IsSignedWithPurpose`.

Reply:
Maybe a dummy question, but that smart contract function, extracts the public key out of the message and signature provided, and checks that the identity has that key. Is that equivalent to verifying that a message has been signed with the key that is claimed to? Is there any security concern in there?

Reply:
Yes, it is the same.   The only difference I see is that `ValidateSignature` implicit assume using the signing key for the check and the `IsSignedWithPurpose` needs explicit the purpose.